src/Operation/InsertOne.php

```diff
@@ -94,6 +105,15 @@ public function __construct(string $databaseName, string $collectionName, $docum
             unset($options['writeConcern']);
         }
 
+        if (isset($options['codec'])) {
```

Comment:
This is processing the document in the constructor (unlike BulkWrite and InsertMany), but I think we should do this before any `$document` validation above.